### ISSUE TITLE
Wrong variable names substitution?

### ISSUE THREAD

**Issue by alex-curiou** [radareorg/radare2#17637]
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Ubuntu 20.04.1
| File format of the file you reverse (mandatory)      | ELF
| Architecture/bits of the file (mandatory)            | arm
| r2 -v full output, **not truncated** (mandatory)     | radare2 4.6.0-git 25649 @ linux-x86-64 git.4.4.0-678-g5ccf9fd48
commit: 5ccf9fd4824ee54e999e92993442ba200c65ab80 build: 2020-09-13__02:36:27

### Expected behavior
variable name in function is
; var int64_t var_0h_2 @ sp+0x20
so the substitution should be
add x29, sp - var_0h_2

### Actual behavior
variable name in function is
; var int64_t var_0h_2 @ sp+0x20
so the substitution should be
add x29, sp - var_0h_2**x20**

### Steps to reproduce the behavior 
Disassemble a binary
![Now](https://user-images.githubusercontent.com/60251601/93007786-83d56300-f575-11ea-9e4f-99cc04b5e9e3.jpg)

### Additional Logs, screenshots, source-code,  configuration dump, ...

Drag and drop zip archives containing the Additional info here, don't use external services or link.

**Comment by ret2libc** [comment]
@alex-curiou Could you provide the exact binary where this happen? It would help reproducing the issue, thanks.

**Comment by alex-curiou** [comment]
Here is the binary, so for example:
r2 -AA libnative-lib.so
s 0x00015780

NOTE: I've reported not correct ABI, so changed from arm to arm64-v8a

[libnative-lib.zip](https://github.com/radareorg/radare2/files/5218360/libnative-lib.zip)

**Comment by trufae** [comment]
yes this is wrong. and this is what IDA does. and this is what i was complaining some days ago because of a change in the disasm. The substituion is wrong because it should even replace the add for a mov to make it meaningful that way.

Obviously having the var that is accessed is nice, but i think it shouldnt be shown that way. It was shown as a comment when using esil before iirc. but its always better to have this replaced inside the instruction. but i dont think this is a good way to render that. And ive been always trying to avoid mimic IDA's disasm which is full of lies like this that make me feel confused when comparing disassemblers

**Comment by alex-curiou** [comment]
That is not a canonical way of showing disassembly, so it is a bit hard to understand and will lead to creation of a our own dialect.
So we will have a bit more closed community ...
From another side, may be some one like it this way and we can have it like an option?

**Comment by trufae** [comment]
An option for what? this disassembly is wrong. it should be fixed. i was stating that IDA is doing things like this, and im against inventing instructions. I dont think r2 community is closed at all. I dont want such thing, not even as an option. because its misleading.

**Comment by ret2libc** [comment]
Could someone provide a snapshot of what is IDA showing exactly?

**Comment by alex-curiou** [comment]
Here is how IDA 7.0 shows that funciton
![IDA](https://user-images.githubusercontent.com/60251601/93123212-6cf75380-f6d0-11ea-866d-485026fb2fc2.jpg)

But I think this way + variables usage in comments is better and more clear and easier to read
![R2](https://user-images.githubusercontent.com/60251601/93123314-8dbfa900-f6d0-11ea-866f-1faf46b5f6a9.jpg)

**Comment by ret2libc** [comment]
Ok, so `add x29, sp, 0x20` is not converted to anything special in IDA either. It is just r2 that incorrectly substitutes things. Then there is definitely a bug in `parse_arm_pseudo.c`. @alex-curiou would you be willing to give this issue a try with a PR?

**Comment by alex-curiou** [comment]
In the coming several weeks I have really busy days ((
if only I suddenly find a solution earlier for the current issue I'm busy with...

**Comment by ret2libc** [comment]
@alex-curiou no problem ;)

I took the liberty to update the description of the issue.

**Comment by alex-curiou** [comment]
Of course,
You are welcome ))

**Comment by trufae** [comment]
i have started to work on fixing this issue but its breaking abi, so wont merge the PRs until we start the abi-breaking stage, there are several more issues related to variable analysis right now, so im trying to collect them to have tests for all of them
